hvantk/datasets/dataset_creation_registry.py

```
@@ -0,0 +1,384 @@
+            Path(self.registry_file).parent.mkdir(parents=True, exist_ok=True)
+
+            data = {
+                dataset_id: self._dataset_entry_to_dict(entry)
+                for dataset_id, entry in self.datasets.items()
+            }
+
+            with open(self.registry_file, 'w') as f:
+                json.dump(data, f, indent=2, default=str)
+            logger.info(f"Saved {len(self.datasets)} datasets to registry")
+        except Exception as e:
+            logger.error(f"Failed to save registry: {e}")
```

Ruff (0.14.0)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)